app/models/ability.rb

```diff
@@ -226,19 +226,16 @@ def to_permission(user)
     # GET    /projects/:project_id/permissions/:id permissions#show {:format=>"json"}
     # DELETE /projects/:project_id/permissions/:id permissions#destroy {:format=>"json"}
 
-    # any user, including guest, with reader permissions on project can access #new
-    can [:new], Permission do |permission|
-      check_model(permission)
-      Access::Core.can?(user, :reader, permission.project)
-    end
 
     # only owners can list, change, or remove permissions
-    can [:index, :show, :create, :destroy], Permission do |permission|
+    can [:show, :create, :update, :destroy], Permission do |permission|
       check_model(permission)
       Access::Core.can?(user, :owner, permission.project)
     end
-  end
 
+     # available to any user, including guest
+     can [:index, :filter, :new], Permission
+  end
```

Comment:
Guests can index and filter permissions route, but not show?

Reply:
I'll have to double check the logic here, but there's generally no reason to restrict access to an endpoint if the endpoint is informational and correctly filters out sensitive results.

It results in nicer to use APIs. A guest can ask to see which permissions they have access too (think loading a page on the client) and the answer should be an empty list rather than an API error.